app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -197,14 +194,12 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
 
     override fun onStop() {
         super.onStop()
-        // This observer initialized in onStart has a reference to fragment's view.
-        // Prevent it leaking the view after the latter onDestroyView.
-        if (openInAppOnboardingObserver != null) {
-            @Suppress("DEPRECATION")
-            // TODO Use browser store instead of session observer: https://github.com/mozilla-mobile/fenix/issues/16949
-            getSessionById()?.unregister(openInAppOnboardingObserver!!)
-            openInAppOnboardingObserver = null
-        }
+
+        openInAppOnboardingObserver?.stop()
+        // This observer is initialized in onStart and holds a reference to
+        // fragment's view. Let's set it to null to prevent leaking the view
+        // after the onDestroyView.
+        openInAppOnboardingObserver = null
```

Comment:
I think this means we need to turn the observer into a `LifecycleFeature` and wrap in our `ViewBound...Thingy` to automatically clear it if the view gets detached? If I remember correctly, we saw the navigation library detach fragments (which destroys the view) and then later attach them again (creating new views) on some navigations and then we leaked the view during that detached state?

Reply:
Afaik the previous code had this problem too, so I think we could also decide to handle this in a follow-up issue.

Reply:
Yes, thought about that too, but wanted to keep separate. I will give it a try and file a follow-up if too much.

Reply:
OK, it was pretty easy and cleaned things up nicely. Updated.